### PR TITLE
Lint tests

### PR DESCRIPTION
**Pull request by islemaster**
Add linting to our test files, with a few special rules defined in test/unit/.eslintrc.js (and documented inline there).  This was super-easy, because our tests were mostly clean already!

### CODE REVIEW

File: test/unit/keycodes.js [hunk 1]
```diff
@@ -17,81 +17,94 @@ describe('_keyCodeFromAlias', function() {
...
-            return char.toUpperCase();
-          } else {
-            return char.toLowerCase();
-          }
-        }).join('');
-
-        expect(_keyCodeFromAlias(upperCaseAlias))
-          .to.equal(_keyCodeFromAlias(lowerCaseAlias))
-          .to.equal(_keyCodeFromAlias(randomCaseAlias));
+        if (KEY.hasOwnProperty(alias)) {
```
Comment (islemaster) [inline test/unit/keycodes.js]: The change here is the `hasOwnProperty` check right inside the `for(...in...)` loop.  There are a few of these.

**Comment by islemaster** [comment]
Note: This PR introduces warnings for TODOs, and there are three in tests that I'm not fixing (visible in the Travis log):
```
/home/travis/build/molleindustria/p5.play/test/unit/input.js
   46:5  warning  Unexpected 'todo' comment  no-warning-comments
   64:5  warning  Unexpected 'todo' comment  no-warning-comments
  110:5  warning  Unexpected 'todo' comment  no-warning-comments
✖ 3 problems (0 errors, 3 warnings)
```

[These TODOs are references](https://github.com/molleindustria/p5.play/blob/master/test/unit/input.js#L46) to issue #43 about key handlers potentially being inaccurate for one frame, and should be removed when that issue is resolved.

File: test/unit/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    // Very frequent: new p5(function () {})
+    "no-empty-function": "off",
+
+    // Chai BDD assertions violate this style rule, so disable it in tests
+    // e.g. expect(value).to.equal('something')
+    "no-unused-expressions": "off",
+
+    // Eventually we want to propagate these four settings up to the main config.
+    "space-before-blocks": "error",
+    "space-before-function-paren": "error",
```
Comment (toolness) [inline test/unit/.eslintrc.js]: WAT?!! But `function foo ()` looks so weird to me... :stuck_out_tongue_winking_eye: 

Ah, it's all good, I'll get over it.

**Comment by toolness** [comment]
Looks good to me, thanks!

File: test/unit/.eslintrc.js (hunk 2 above)
Comment (islemaster) [inline test/unit/.eslintrc.js]: Oh, I'm not sure this is what I meant it to me.  I propose updating this to use the setting [`{anonymous: 'always', named: 'never'}`](http://eslint.org/docs/rules/space-before-function-paren#anonymous-always-named-never).

```javascript
// Required
function foo() {};
var foo = function () {};

// Broken
function foo () {};
var foo = function() {};
```

Rationale being if you put parens adjacent to a keyword (`function()`) you should do it everywhere (`if()`, `for()`). Objections?

Comment (islemaster) [inline test/unit/.eslintrc.js]: Never mind - we _almost_ follow the "never" version of this rule in p5.play.js, I'm going to switch to that.

**Pull request merged**